src/Filter/FilterInterface.php

```diff
@@ -27,144 +27,107 @@ interface FilterInterface
     /**
      * Apply the filter to the QueryBuilder instance.
      *
-     * @param string  $alias
-     * @param string  $field
      * @param mixed[] $value
      */
-    public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $value);
+    public function filter(ProxyQueryInterface $queryBuilder, string $alias, string $field, $value): void;
 
     /**
      * @param mixed $query
      * @param mixed $value
      */
-    public function apply($query, $value);
+    public function apply($query, $value): void;
 
-    /**
-     * Returns the filter name.
-     *
-     * @return string
-     */
-    public function getName();
+    public function getName(): string;
 
-    /**
-     * Returns the filter form name.
-     *
-     * @return string
-     */
-    public function getFormName();
+    public function getFormName(): string;
 
-    /**
-     * Returns the label name.
-     *
-     * @return string|bool
-     */
-    public function getLabel();
+    public function getLabel(): ?string;
 
-    /**
-     * @param string $label
-     */
-    public function setLabel($label);
+    public function setLabel(string $label): void;
 
     /**
-     * @return array
+     * @return array<string, mixed>
      */
-    public function getDefaultOptions();
+    public function getDefaultOptions(): array;
 
     /**
-     * @param string     $name
-     * @param mixed|null $default
+     * @param mixed $default
      *
      * @return mixed
      */
-    public function getOption($name, $default = null);
+    public function getOption(string $name, $default = null);
 
     /**
-     * @param string $name
-     * @param mixed  $value
+     * @param mixed $value
      */
-    public function setOption($name, $value);
+    public function setOption(string $name, $value): void;
 
     /**
-     * @param string $name
+     * @param array<string, mixed> $options
      */
-    public function initialize($name, array $options = []);
+    public function initialize(string $name, array $options = []): void;
 
     /**
-     * @return string
+     * @throws \RuntimeException
      */
-    public function getFieldName();
+    public function getFieldName(): string;
 
     /**
      * @return array<string, string> array of mappings
      */
-    public function getParentAssociationMappings();
+    public function getParentAssociationMappings(): array;
 
     /**
+     * @throws \RuntimeException
```

Comment:
I think we could add an explanation about when this exception can occur. Same for others

Reply:
you mean like `@throws Exception this happens when ...` syntax? I will add them

Reply:
Yes indeed, I was meaning `@throws \RuntimeException when there is no foo`.

Reply:
Done for the 3 exceptions